### PR TITLE
[IOTDB-3460] Python API: Add support for DBAPI

### PR DESCRIPTION
**Pull request by noorall**
I made the following changes in this PR:

1. Implement the methods specified in DBAPI2.0
2. Add test code for the DBAPI module

**JIAR:** https://issues.apache.org/jira/browse/IOTDB-3460
**DBAPI2.0:** https://peps.python.org/pep-0249/

### CODE REVIEW

**Comment by noorall** [comment]
> 

I will add some documents soon.

**Comment by noorall** [comment]
> Good job! Would you like to add some user guide documents?

User guide documents has been added.

File: client-py/README.md [hunk 1]
```diff
@@ -353,6 +353,60 @@ class MyTestCase(unittest.TestCase):
 by default it will load the image `apache/iotdb:latest`, if you want a specific version just pass it like e.g. `IoTDBContainer("apache/iotdb:0.12.0")` to get version `0.12.0` running.
 
 
+### IoTDB DBAPI
+
+IoTDB DBAPI implements the Python DB API 2.0 specification (https://peps.python.org/pep-0249/), which defines a common
+interface for accessing databases in Python.
```
Comment (HTHou) [inline client-py/README.md:359]: And consider adding this to https://github.com/apache/iotdb/blob/master/docs/zh/UserGuide/API/Programming-Python-Native-API.md and https://github.com/apache/iotdb/blob/master/docs/UserGuide/API/Programming-Python-Native-API.md

Comment (noorall) [reply client-py/README.md:359]: added.

**Review by HTHou — APPROVED** [review]

**Pull request merged**